sycl/source/detail/kernel_compiler/kernel_compiler_opencl.cpp

```diff
@@ -0,0 +1,197 @@
+//==-- kernel_compiler_ opencl.cpp  OpenCL kernel compilation support      -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <sycl/detail/pi.hpp> // getOsLibraryFuncAddress
+#include <sycl/exception.hpp> // for make_error_code
+
+#include "kernel_compiler_opencl.hpp"
+
+#include "../online_compiler/ocloc_api.h"
+
+#include <cstring> // strlen
+#include <numeric> // for std::accumulate
+
+namespace sycl {
+inline namespace _V1 {
+namespace ext::oneapi::experimental {
+namespace detail {
+
+// copy/pasta from online_compiler.cpp
```

Comment:
It seems that most of this code is present in online_compiler.cpp. Can we avoid duplicating the code? I believe it will be better to move common code to some common file.

Reply:
I understand that online_compiler.cpp will be removed at some point. But it will not happen immediately. And for that time we will have to maintain two copies of the same code.

Reply:
Seems that this comment was missed. What do you think about this?

Reply:
I was going to move this to a common header, when I ran into some issues.   I think I'll remove the copy/paste comment. This function `checkOclocLibrary` doesn't appear in the online_compiler, just some of the code here does. They throw different exceptions.  And due to lifecycle difference, this one operates on a static var, the one in the online_compiler does not.  Unifying the code is not worth the effort, especially for a deprecated component.